packages/protocol/test/common/accounts.ts

```diff
@@ -432,148 +379,469 @@ contract('Accounts', (accounts: string[]) => {
     })
   })
 
-  Object.keys(authorizationTestDescriptions).forEach((key) => {
-    describe('authorization tests:', () => {
-      let authorizationTest: any
-      beforeEach(async () => {
-        authorizationTest = authorizationTests[key]
-        await accountsInstance.createAccount()
-      })
+  describe('generic authorization', () => {
+    const account2 = accounts[1]
+    const signer = accounts[2]
+    const signer2 = accounts[3]
+    const role = keccak256('Test Role')
+    const role2 = keccak256('Test Role 2')
+    let sig
+
+    beforeEach(async () => {
+      sig = await kit.connection.signTypedData(
+        signer,
+        buildAuthorizeSignerTypedData({
+          account,
+          signer,
+          role,
+          accountsContractAddress: accountsInstance.address,
+          chainId,
+        })
+      )
+      await accountsInstance.createAccount()
+      await accountsInstance.createAccount({ from: account2 })
+    })
+
+    it('should recover the correct signer', async () => {
+      const recoveredSigner = await accountsInstance.getRoleAuthorizationSigner(
+        account,
+        signer,
+        role,
+        sig.v,
+        sig.r,
+        sig.s
+      )
+      expect(signer).to.be.equal(recoveredSigner)
+    })
 
-      describe(`#authorize${upperFirst(authorizationTestDescriptions[key].subject)}()`, () => {
-        const authorized = accounts[1]
-        let sig
+    it('should set the authorized signer in two steps', async () => {
```

Comment:
I think you might want to have a separate test to check that one `completeSignerAuthorization` reverts when `authorizeSigner` wasn't called yet? In general I'm wondering if you want to split this group into the conventional `describe('#method')` groups that test reversion and event emission separately for readability?